[Java Sync] Shift Java ToC (#615)
mongodb/docs-java: source/versioning.txt

==========
Versioning
==========

.. meta::

>    :description: Learn how to use versioning in the {+driver-long+}.

Reviewer: S: "how to use" seems a little off to me, maybe: 
```suggestion
   :description: Learn about versioning in the {+driver-long+}.
```
